src/gui/instrument/LfoGraph.cpp

```diff
@@ -78,81 +74,113 @@ void LfoGraph::paintEvent(QPaintEvent*)
 	const f_cnt_t attackFrames = params->getLfoAttackFrames();
 	const f_cnt_t oscillationFrames = params->getLfoOscillationFrames();
 	const bool x100 = params->getX100Model().value();
-	const int waveModel = params->getLfoWaveModel().value();
+	const int lfoWaveModel = params->getLfoWaveModel().value();
+	const auto * userWave = params->getLfoUserWave().get();
 
-	int LFO_GRAPH_W = m_lfoGraph.width() - 3;	// subtract border
-	int LFO_GRAPH_H = m_lfoGraph.height() - 6;	// subtract border
-	int graph_x_base = 2;
-	int graph_y_base = 3 + LFO_GRAPH_H / 2;
+	const int margin = 3;
+	const int lfoGraphWidth = width() - margin; // subtract margin
+	const int lfoGraphHeight = height() - 2 * margin; // subtract margin
+	int graphBaseX = 2;
+	int graphBaseY = margin + lfoGraphHeight / 2;
 
-	const float frames_for_graph =
+	const float framesForGraph =
 		SECS_PER_LFO_OSCILLATION * Engine::audioEngine()->baseSampleRate() / 10;
 
-	const float gray = 1.0 - fabsf(amount);
-	const auto red = static_cast<int>(96 * gray);
-	const auto green = static_cast<int>(255 - 159 * gray);
-	const auto blue = static_cast<int>(128 - 32 * gray);
-	const QColor penColor(red, green, blue);
-	p.setPen(QPen(penColor, 1.5));
+	float oscFrames = oscillationFrames * (x100 ? 100. : 1.);
 
-	float osc_frames = oscillationFrames;
+	QPolygonF polyLine;
+	polyLine << QPointF(graphBaseX - 1, graphBaseY);
 
-	if (x100) { osc_frames *= 100.0f; }
-
-	float old_y = 0;
-	for (int x = 0; x <= LFO_GRAPH_W; ++x)
+	// Collect the points for the poly line by sampling the LFO according to its shape
+	for (int x = 0; x <= lfoGraphWidth; ++x)
 	{
-		float val = 0.0;
-		float cur_sample = x * frames_for_graph / LFO_GRAPH_W;
-		if (static_cast<f_cnt_t>(cur_sample) > predelayFrames)
+		float value = 0.0;
+		float currentSample = x * framesForGraph / lfoGraphWidth;
```

Comment:
I believe this is `const` ?

Reply:
`currentSample` is mutated in line 102 and therefore cannot be made `const`.

Reply:
Oh! It's inline that's why I completely missed it.